### DISCUSSION TITLE
Save BertTopic model to GCS

### DISCUSSION THREAD

**Discussion by MourouzidouElisa** [MaartenGr/BERTopic#878] [Q&A]
Hi, I would like to move my BertTopic Model to my google cloud storage. I've saved the model using: topic_model.save("model_one"), I downloaded the file and then uploaded to my GCS but when I do that I cannot use the model (I have a "null" error when I run it). I am trying to find another way to save the model hoping that I'll be able to uploaded and then uploaded it to my GCS after. Is there any other way you suggest to save BertTopic?

Thank you in advance :)

**Comment by MaartenGr** [comment]
Could you share the exact code for training the model and then loading it? Also, could you share the full error, that will help identify what exactly is happening? Lastly, did you make sure that the environments between saving and loading the model have the same python versions and dependencies?

**Reply by MourouzidouElisa** [reply]
Thank you for your reply 

Here is my code:
![Annotation 2022-12-18 172050](https://user-images.githubusercontent.com/80224661/208310955-736dabae-ef9f-4cbe-a960-aa2641691b94.png)

and here is the error : 

![image](https://user-images.githubusercontent.com/80224661/208311036-9797fbd7-6e54-49e5-8343-247b6cfb3161.png)

**Reply by MaartenGr** [reply]
Based on the error message, it seems that there are two additional `"_"` used when loading the model:

```python
topic_model.BERTopic.load("BertTopicModel__")
```

Perhaps there is some hidden string ending up in the name.

**Reply by MourouzidouElisa** [reply]
ah yes sorry, I changed the name afterwards. The screenshot is wrong but the error is the same.

**Reply by MaartenGr** [reply]
Ah okay, did you also make sure that the python and dependency versions are exactly the same between the environment where you saved the model and where you loaded the model?

**Reply by MourouzidouElisa** [reply]
Yes, I make sure the environments are the same. I am thinking that the issue is caused when I download the model. What I am doing is: I save the model, I download it from the platform and then upload it again to the same platform but in different instance and then load the model there. Is there any chance that I ruin the model while downloading/uploading it?

**Reply by MaartenGr** [reply]
Strange, this should not happen if the environments are the same, including whether GPU is enabled or not, etc. I would be quite surprised if downloading and uploading the model would do something to the file. If the file was correctly uploaded, then there should be no problem. Are the sizes of the downloaded and uploaded files the same? Also, it might be worthwhile to re-install BERTopic from a fresh environment to see if that helps. One other thing, I would be a bit surprised but perhaps using python 3.8 might solve your issue.

**Reply by MourouzidouElisa** [reply]
Ah, the sizes of the initial saved model and the downloaded one are not the same. I dont know why though. I'll probably have to find another way to move the model.
